---
title: "mittermeieri-lehilahytsara exploration"
date: "`r Sys.Date()`"
output:
   bookdown::html_document2:
    toc: true
    css: style.css
---

```{r setup-cache, include=FALSE}
my_cache <- FALSE

```

```{r setup, include=FALSE, cache=my_cache}
suppressMessages(library(knitr))
suppressMessages(library(here))

opts_chunk$set(
  echo = FALSE,
  include = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.path = here('fig/markdown/'),
  dev = "svg" # output file in SVG format
  ) 

library(tidyverse)
library(vcfR) # vcf handling
library(gdsfmt) # vcf handling
library(SNPRelate) # vcf handling
library(sf) # handling geographical data
library(rnaturalearth) # base map
library(ggspatial) # scale bar
library(osmdata) # OSM coastline
library(osmplotr) # get_bbox()
library(patchwork) # plot placement
library(cowplot) # plot placement
#library(raster) # plot placement
library(ggrepel) # repelling labels in plots
library(ggnewscale) # multiple color/fill scales in plots
library(ggforce) # group labels in plots
library(kableExtra)
library(plotly)
```

```{r setup_metadata, cache=my_cache}
## Variables:
setID <- 'mitleh.nooutgroup.mac3.FS6'
focal_sp <- c('mmit', 'mleh')

## Input files:
infile_vcf <- here('seqdata/vcf/stacks', paste0(setID, '.vcf.gz'))
infile_lookup <- here('metadata/radseq_metadata_link/lookup_IDshort.txt')
infile_pops <- here('metadata/lemurs_metadata_link/lookup/sites_pops.txt')
infile_cols <- here('metadata/lemurs_metadata_link/lookup/colors.txt')

## Output files:
file_snpgds <- here('analyses/PCA/gds_files/', paste0(setID, '.gds'))

## Check VCF to get individuals that passed filtering:
if(!file.exists(file_snpgds)) {
  snpgdsVCF2GDS(infile_vcf, file_snpgds, method = "biallelic.only")
}
snps <- snpgdsOpen(file_snpgds, allow.duplicate = TRUE)
inds_pass <- read.gdsn(index.gdsn(snps, "sample.id"))

## Metadata files:
lookup_raw <- read.delim(infile_lookup, as.is = TRUE) %>%
  filter(sp %in% focal_sp,
         !ID %in% c('mleh034', 'mleh035')) %>%
  select(ID, Sample_ID, species, sp, site)

pops <- read.csv(infile_pops, as.is = TRUE) %>%
  dplyr::filter(sp %in% focal_sp) %>%
  dplyr::mutate(pop2 = ifelse(is.na(pop2), sp, pop2)) %>%
  dplyr::select(site, site_short, site_lump, pop2)

cols <- read.delim(infile_cols, as.is = TRUE) %>%
  dplyr::select(pop, col)

## Popgroup labels and factor orders:
pop2_order <- c('mmit', 'leh_N', 'leh_hi', 'leh_S')
pop2_labs <- c('mittermeieri', 'lehi: north', 'lehi: high', 'lehi: south')
pops <- pops %>%
  mutate(pop2 = fct_relevel(pop2, pop2_order),
         pop_group = factor(recode(pop2,
                                   mmit = "mittermeieri",
                                   leh_N = "lehi: north",
                                   leh_hi = 'lehi: high',
                                   leh_S = 'lehi: south')))

## Merge metadata files:
lookup <- lookup_raw %>%
  merge(., pops, by = 'site', all.x = TRUE) %>%
  merge(., cols, by.x = 'pop2', by.y = 'pop', all.x = TRUE) %>%
  rename(col_pop2 = col) %>%
  merge(., cols, by.x = 'sp', by.y = 'pop', all.x = TRUE) %>%
  rename(col_sp = col) %>%
  mutate(filtering = ifelse(ID %in% inds_pass, 'pass', 'fail')) %>%
  select(ID, Sample_ID, sp, site_short, site_lump,
                pop2, pop_group, col_pop2, col_sp, filtering)
```

In the msp3 manuscript, we only included _lehilahytsara_ samples from an area that is well north of the published IUCN range of the species, and some of which were originally labelled as _mittermeieri_.

So a question relevant to their (lack of) delimitation as species is what differentiation looks like when more southerly _lehilahytsara_ populations are also included.

## Maps

The map below shows the distributions of _mittermeieri_ (green) and _lehilahytsara_ (blue). IUCN distibutions have a darker color and a solid contour line; hand-drawn estimated "additional" distributions (based on our sampling sites and forest cover) have a lighter color and dotted contour lines.

```{r dist_map, fig.align = 'center', out.width = "400px", cache=my_cache}
fig_dist <- here('map/map_dist_inset.png')
include_graphics(fig_dist)
```

The maps below shows the sites for which we have samples with RADseq data, colored by species (with remaining forest in the background).
On the right, some sites that are very close to each other have been lumped together (e.g., "Anjiahely+" consists of "Anjiahely" and "Antsahabe") -- to avoid cluttering, I will use these lumped site names when visualizing PCA and ADMIXTURE results below.

```{r site_map1, fig.align = 'center', out.width = "500px", cache=my_cache}
fig_dist <- here('map/map_sites_sp.png')
include_graphics(fig_dist)
```

Based on locations and genetic differentiation (see results below), I have divided the _lehilahytsara_ samples into three "population groups": "lehi-north", "lehi-south", and "lehi-high" (highland plateau sites, which is only Ankafobe after filtering). See the sites maps below, now colored by population group.
**For the msp3 paper, we only included _lehilahytsara_ samples from the "lehi-north" population group (Riamalandy and the Ambavala area).**

```{r site_map2, fig.align = 'center', out.width = "500px", cache=my_cache}
fig_dist <- here('map/map_sites_pop2.png')
include_graphics(fig_dist)
```

Finally, here is a map simply showing sampling sites by population group, with vegetation types of remaining vegetation shown:

```{r pop_map,  fig.align = 'center', out.width = "600px", cache=my_cache}
fig_dist <- here('map/map_pops.png')
include_graphics(fig_dist)
```

## RADseq metadata
- Samples were genotyped with Stacks and then filtered using a fairly stringent filtering procedure which removes both sites and samples. In the tables below,
the "pass" indicates samples that passed this filtering, and "fail" indicates samples that failed this filtering step. (Not all samples that "failed" this are useless though, given the stringency of the filtering.) I did not include the two or so samples from captive colonies.

- **Number of samples by population group ("pop_group"):**
```{r metadata_summary_1, results='show'}
by_popgroup <- lookup %>%
  group_by(pop_group, filtering) %>%
  summarise(count = n()) %>%
  pivot_wider(names_from = filtering, values_from = count) %>%
  ungroup() %>%
  mutate(total = fail + pass) %>%
  select(pop_group, pass, fail, total) %>%
  as.data.frame()
kable(by_popgroup) %>% kable_styling(full_width = F)
```

- **Number of samples by site:**
```{r metadata_summary_2, results='show'}
by_site <- lookup %>%
  group_by(site_short, filtering) %>%
  summarise(count = n()) %>%
  pivot_wider(names_from = filtering, values_from = count,
              values_fill = list(count = 0)) %>%
  ungroup() %>%
  mutate(total = fail + pass) %>%
  merge(., select(pops, site_short, pop_group),
        by = 'site_short', all.x = TRUE) %>%
  select(site_short, pass, fail, total, pop_group) %>%
  as.data.frame()
kable(by_site) %>%
  kable_styling(full_width = F) %>% 
  scroll_box(width = "400px", height = "300px")
```

- **List of all samples:**
```{r metadata_summary_3, results='show'}
allsamples <- lookup %>%
  arrange(pop_group, site_lump, ID) %>%
  select(ID, Sample_ID, sp, site_short, site_lump, pop_group, filtering)
kable(allsamples) %>%
  kable_styling(full_width = F) %>% 
  scroll_box(width = "600px", height = "300px")
```

## Jordi's tree
Here is part of Jordi's tree that he sent around a while ago. This seems to suggest that the main axis of differentiation is between _mittermeieri_ + "lehi-north" on one hand versus "lehi-south" on the other (note that no samples from "lehi-high" were included). At its most extreme, this could mean that one could in fact distinguish two species, but that "lehi-north" would just have to be included with _mittermeieri_...

```{r jordi_tree, out.width = "600px", fig.align = 'center', }
fig_tree <- here('fig/jordi/mittlehi_raxml_ed.png')
include_graphics(fig_tree)
```

## Splitstree
Below is an annotated Splitstree network, with sites and population groups labelled. This is roughly in line with Jordi's tree, but it's fairly messy, and might rather suggest an isolation-by-distance (IBD) pattern that would show even less support for distinct groups if geographically intermediate populations were included.


```{r splitstree, fig.align = 'center', out.width = "600px", cache=my_cache}
fig_splitstree <- here('analyses/splitstree/fig/mitleh.nooutgroup.mac3.FS6_fig.png')
include_graphics(fig_splitstree)
```

## ADMIXTURE
Below is an ADMIXTURE plot showing cross-validation errors at different values of K.
This suggests that the fit gets worse with a higher number of clusters, already slightly so going from K=1 -> K=2. So no support for two species.

```{r admixture_1, results="hide", fig.align='center', fig.width=5, fig.height=4, out.width="400px"}
## Libraries and scripts:
library(gridExtra)
library(grid)
library(RColorBrewer)
library(ggpubr)
source(here("scripts/genomics_link/admixture/admixture_plot_fun.R"))

## Setup:
filedir <- here('analyses/admixture/output/')
lookup <- lookup %>%
  mutate(ID2 = sub(':m', ':', paste0(site_lump, ':', ID)),
         pop2 = fct_relevel(pop2, pop2_order)) %>%
  arrange(pop2) %>%
  mutate(col_pop2 = fct_inorder(as.factor(col_pop2)))
    
## Colors:
barcol_pal <- brewer.pal(name = 'Set2', n = 8)
bgcol_pal <- levels(lookup$col_pop2)

## Group labels:
pop2_labs <- c(leh_N = 'lehi: north',
               leh_S = 'lehi: south',
               leh_hi = 'lehi: high',
               mmit = 'mittermeieri')
grouplab.labeller <- labeller(pop2 = pop2_labs)

## CV-plot:
kp <- k.plot(setID)
print(kp)
```

Below is an ADMIXTURE plot with K=2, K=3, and K=4.
At K=2, "lehi-north" is grouped with _mittermeieri_, like in Jordi's tree.
At K=3, "lehi-high" gets its own cluster.
At K=4, each of the population groups gets its own cluster.

```{r admixture_2_prep, results="hide", include=FALSE}
k2 <- Qdf(setID, K = 2, sort_by = 'site_lump', inds_df = lookup) %>%
  ggax.v(., indID.column = 'ID2', group.column = 'pop2',
         barcols = barcol_pal, indlabs = FALSE, ylab = 'K=2',
         grouplab.bgcol = bgcol_pal, grouplab.labeller = grouplab.labeller)
k3 <- Qdf(setID, K = 3, sort_by = 'site_lump', inds_df = lookup) %>%
  ggax.v(., indID.column = 'ID2', group.column = 'pop2',
         barcols = barcol_pal, indlabs = FALSE, ylab = 'K=3',
         grouplab.bgcol = bgcol_pal, grouplab.labeller = grouplab.labeller)
k4 <- Qdf(setID, K = 4, sort_by = 'site_lump', inds_df = lookup) %>%
  ggax.v(., indID.column = 'ID2', group.column = 'pop2',
         barcols = barcol_pal, indlabs = TRUE, ylab = 'K=4',
         grouplab.bgcol = bgcol_pal, grouplab.labeller = grouplab.labeller)
p_admix <- ggarrange(k2, k3, k4,
                     ncol = 1, nrow = 3, heights = c(0.4, 0.4, 1))
```

```{r admixture_2, results="hide", fig.align='center', out.width = "750px"}
print(p_admix)
```

## PCA
Below is a PCA plot, with ellipses around each (lumped) site, and a different point shape for each population group. The inset just shows colors by population group.
I'd say these results are also consistent with a rough IBD pattern going from "lehi-south" -> "lehi-north"" -> _mittermeieri_, but with the highland population group relatively distinct. 

```{r PCA, include=FALSE, cache=my_cache}
## Prep:
source(here('scripts/genomics_link/PCA/PCA_R_fun.R'))

## Run PCA & process results:
pca_raw <- snpgdsPCA(snps, autosome.only = FALSE, num.thread = 1)
pca <- process_pca(pca_raw, my_lookup = lookup) %>%
  mutate(pop2 = factor(pop2, levels = pop2_order)) %>%
  arrange(pop2) %>%
  mutate(col_pop2 = fct_inorder(as.factor(col_pop2)))
eig <- data.frame(PC = 1:length(pca_raw$eigenval), eig = pca_raw$eigenval)
eig <- eig[complete.cases(eig), ]

## Main plot:
p1 <- pcplot(pca, eigenvals = eig$eig, strokesize = 1,
             xmin_buffer = 0.2, xmax_buffer = 0.1,
             ymin_buffer = 0.2, ymax_buffer = 0.2,
             shape_by = 'pop2', shape_by_name = 'Population:',
             shape_by_labs = pop2_labs, shapes = c(8, 15, 17, 18),
             col_by = 'site_lump', axis_title_size = 14,
             draw_boxes = TRUE, boxlabsize = 7) +
  theme(plot.margin = margin(0.3, 0.6, 0.3, 0.3, "cm"),
        legend.justification = "top",
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 12, face = 'bold'))

## Inset plot:
p2 <- pcplot(pca, eigenvals = eig$eig, strokesize = 1,
             xmin_buffer = 0.01, xmax_buffer = 0.01,
             ymin_buffer = 0.35, ymax_buffer = 0.01,
             col_by = 'pop_group', col_by_name = 'Population:',
             col_by_labs = levels(pca$pop_group),
             cols = levels(pca$col_pop2)) +
  guides(color = guide_legend(nrow = 2)) +
  labs(title = 'Color by pop. group:') +
  theme(plot.title = element_text(size = 12, hjust = 0.5),
        axis.ticks = element_blank(),
        axis.title = element_blank(),
        legend.position = c(0.5, 0.1),
        legend.text = element_text(size = 9),
        legend.title = element_blank(),
        legend.box.background = element_rect(colour = 'grey20', size = 1),
        legend.margin = margin(c(0, 0, 0, 0)),
        legend.spacing.y = unit(0, 'cm'),
        legend.key.height = unit(0.15, 'cm'),
        legend.box.spacing = unit(0, 'cm'),
        legend.box.margin = margin(c(0, 0, 0, 0)),
        plot.margin = margin(0.1, 0.2, 0, 0, "cm"))

p_pca <- ggdraw() + draw_plot(p1) +
  draw_plot(p2, x = 0.65, y = 0.02, width = 0.32, height = 0.45)
```

```{r PCA_plot, fig.align = 'center', out.width = "750px", cache=my_cache}
print(p_pca)
```

## IBD
- Genetic versus geographic distances by population, with "interspecific" ( _lehilahytsara_ versus _mittermeieri_) comparisons in red and intraspecific comparisons in blue. The highest genetic differentiation _is_ interspecific but also at the largest geographic distances. There is no big jump like we saw for msp3 versus _macarthuri_.
```{r IBD, results = 'show', out.width = "700px", fig.align = 'center', cache=my_cache}
IBD_fig <- paste0(here('analyses/IBD/fig/mitleh.nooutgroup.mac3.FS6_mitleh.png'))
include_graphics(IBD_fig)
```

```{r IBD-prep, results='hide', include=FALSE, cache=my_cache}
#### SET-UP --------------------------------------------------------------------
source(here('scripts/genomics_link/IBD/IBD_fun.R'))
input_dir <- here('analyses/IBD/RDS/')
output_dir <- here('analyses/IBD/fig/')
my_cols <- RColorBrewer::brewer.pal(name = 'Set1', n = 3)[1:2]

p_mitleh <- ibd_plot_wrap(setID, subsetID = 'mitleh', input_dir = input_dir,
                          my_cols = my_cols)
pl <- p_mitleh +
  aes(text = paste0(site1, '(', sp1, ') - ', site2, '(', sp2, ')'))
```

- Interactive version of the IBD plot:
```{r IBD-interactive, results='show', include=TRUE, out.width = "900px", fig.align = 'center', cache=my_cache}
ggplotly(pl)
```

## Concluding
- I'd say this doesn't support two (or more) species, so in that sense we're on solid ground if we want to recommend to lump _mittermeieri_ and _lehilahytsara_ in the msp3 manuscript.
